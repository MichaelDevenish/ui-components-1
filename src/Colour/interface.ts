export interface ColourProps {
  description: string
}

export interface SProps extends ColourProps {
  hex: string
}

export interface ColourObject {
  name: string;
  hex: string;
}

export interface ColourWithNameAndValue {
  [key: string]: string
}
