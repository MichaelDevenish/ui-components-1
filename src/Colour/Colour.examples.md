### A Colour Component
```jsx
const { brandColours } = require('./Colour');

<Colour hex={brandColours.intelliRoyalBlue} description="intelli-royal-blue" />
```
