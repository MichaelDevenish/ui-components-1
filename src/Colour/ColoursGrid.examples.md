### Brand Colours
```jsx
const { brandColoursArray } = require('./Colour');

<ColoursGrid colours={brandColoursArray} />
```

### App Colours
```jsx
const { appColoursArray } = require('./Colour');

<ColoursGrid colours={appColoursArray} />
```