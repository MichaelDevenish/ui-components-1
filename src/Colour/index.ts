export {
  Colour,
  brandColours,
  appColours
} from './Colour'

export {
  ColoursGrid
} from './ColoursGrid'
