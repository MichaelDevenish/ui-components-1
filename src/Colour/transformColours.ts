import { kebabCase } from 'lodash'

export const transformColours = (c: any) => Object.keys(c).map(k => ({
  name: kebabCase(k),
  hex: c[k]
}))
