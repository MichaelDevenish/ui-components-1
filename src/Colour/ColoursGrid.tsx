import React from 'react'
import styled from 'styled-components'
import { ColourObject, Colour, ColourWithNameAndValue } from './Colour'

export interface ColourGridProps {
  colours: ColourObject[]
}

export class ColoursGrid extends React.PureComponent<ColourGridProps> {
  render () {
    const WrappedList = styled.div`
      display: flex;
      flex-wrap: wrap;
      max-width: 50rem;
      margin-left: 0;
    `
    const { colours } = this.props

    return (
      <WrappedList>
        {colours.map(c => <Colour hex={c.hex} description={c.name} key={c.name} />)}
      </WrappedList>
    )
  }
}
