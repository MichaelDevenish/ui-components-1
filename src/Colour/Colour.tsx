import React, { Fragment } from 'react'
import styled from 'styled-components'
import { transformColours } from './transformColours'

export interface ColourProps {
  description: string
}

export interface SProps extends ColourProps {
  hex: string
}

export interface ColourObject {
  name: string
  hex: string
}

export interface ColourWithNameAndValue {
  [key: string]: string
}

export const brandColours: ColourWithNameAndValue = {
  intelliRoyalBlue: '#432DF3',
  intelliGreen: '#04DBAC',
  intelliYellow: '#FFDD37',
  intelliRed: '#FF4D50',
  intelliCyan: '#03DEEB',
  intelliPurple: '#A35AFF',
  intelliCharcoal: '#303035',
  intelliGrey1: '#B3B3B5',
  intelliGrey2: '#E4E4E5',
  intelliBaseWhite: '#F5F7F9',
  intelliOffWhite: '#F5F5F5'
}

export const appColours: ColourWithNameAndValue = {
  intelliAppBlue: '#0F68FA',
  intelliDarkGrey: '#47525D',
  intelliGreyBlue1: '#C3CCD0',
  intelliGreyBlue2: '#DFE5E8',
  intelliWhite: '#FFFFFF',
  intelliRoyalBlueLight: '#EBE9FE',
  intelliAppBlueLight: '#D7F2FF',
  intelliGreenLight: '#E0FEF8',
  intelliYellowLight: '#FFF9DA',
  intelliRedLight: '#FFE6E6'
}

/**
 * This component is for demonstration of the colour plates purpose
 */
export class Colour extends React.PureComponent<SProps> {
  render () {
    const { hex, description } = this.props
    const Box = styled.div`
      width: 10em;
      height: 7.5em;
      background-color: ${hex};
      margin-bottom: 1em;
    `
    const ColourName = styled.p`
      font-weight: 600;
    `
    return (<div>
      <Box />
      <ColourName> {description} </ColourName>
      <p> {hex} </p>
    </div>)
  }
}

export const brandColoursArray: ColourObject[] = transformColours(brandColours)
export const appColoursArray: ColourObject[] = transformColours(appColours)
