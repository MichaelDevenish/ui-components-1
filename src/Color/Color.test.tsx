import { mount } from 'enzyme'
import React from 'react'
import { brandColors } from './Colour'
import { Color } from './'
import 'jest-styled-components'

describe('<Colour />', () => {
  it(`should render a colour component`, () => {
    const wrapper = mount(
      <Colour hex={brandColours.intelliRoyalBlue} description='intelli-royal-blue' />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
