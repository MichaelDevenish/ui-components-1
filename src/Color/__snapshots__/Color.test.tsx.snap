// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> should render a color component 1`] = `
.c0 {
  width: 10em;
  height: 7.5em;
  background-color: #432DF3;
  margin-bottom: 1em;
}

.c1 {
  font-weight: 600;
}

<Color
  description="intelli-royal-blue"
  hex="#432DF3"
>
  <div>
    <styled.div
      hex="#432DF3"
    >
      <div
        className="c0"
      />
    </styled.div>
    <styled.p>
      <p
        className="c1"
      >
         
        intelli-royal-blue
         
      </p>
    </styled.p>
    <p>
       
      #432DF3
       
    </p>
  </div>
</Color>
`;
